src/MongoDB/Driver/ReadPreference.cpp

```diff
@@ -85,9 +85,27 @@ void HHVM_METHOD(MongoDBDriverReadPreference, _setReadPreferenceTags, const Arra
 	}
 }
 
+void HHVM_METHOD(MongoDBDriverReadPreference, _setMaxStalenessMS, int maxStalenessMS)
+{
+	MongoDBDriverReadPreferenceData* data = Native::data<MongoDBDriverReadPreferenceData>(this_);
+
+	/* Validate that maxStalenessMS is not used with PRIMARY readPreference */
+	if (mongoc_read_prefs_get_mode(data->m_read_preference) == MONGOC_READ_PRIMARY) {
+		throw MongoDriver::Utils::throwInvalidArgumentException("maxStalenessMS may not be used with primary mode");
+	}
+
+	mongoc_read_prefs_set_max_staleness_ms(data->m_read_preference, maxStalenessMS);
+
+	if (!mongoc_read_prefs_is_valid(data->m_read_preference)) {
+		/* Throw exception */
+		throw MongoDriver::Utils::throwInvalidArgumentException("Read preference is not valid");
```

Comment:
I can't call c functions in the constructors as it's implemented in PHP.

mongoc_read_prefs_is_valid() is also run when setting tagsets through _setReadPreferenceTags.